### PR TITLE
Add page for test results

### PR DESCRIPTION
**Pull request by frankieroberto**
This adds a new view for "test results", and links to this view from the relevant entries in the activity timeline.

In future it will also be linked to from a "Supporting information" tab.

## Checklist:
- [x] Have you documented your changes in the pull request description?
- [x] Does the change present any security considerations?
- [ ] Has acceptance criteria been tested by a peer?
- [ ] Has the CHANGELOG been updated? (If change is worth telling users about.)

#### Before

Activity timeline links directly to the test result file:

Screenshot 2020-06-01 at 10 46 32

#### After

Activity timeline links to new page for the test result:

Screenshot 2020-06-01 at 10 46 43

New test result page:

Screenshot 2020-06-01 at 16 16 48

### CODE REVIEW

File: psd-web/app/models/audit_activity/test/result.rb [hunk 1]
```diff
@@ -13,6 +13,18 @@ def email_update_text(viewer = nil)
...
+  #
+  # This is a hack, as there is currently no direct association between the
+  # AuditActivity record and the test result record it is about. So the only
+  # way to retrieve this is by relying upon our current behaviour of attaching the
+  # same actual file to all of the AuditActivity, Investigation and Test records.
+  def test_result
+    attachment.blob.attachments
+      .find_by(record_type: "Test")
+      .record
+  end
```
Comment (frankieroberto) [inline psd-web/app/models/audit_activity/test/result.rb:26]: We can hopefully remove this in future by directly associating the audit activity record with the thing it is about (eg test result, corrective actions, etc). Currently these associations are present for _some_ types of activity record, via a bunch of optional foreign keys (eg `correspondance_id`) but not others.

File: psd-web/app/views/investigations/activities/_activity_card.html.erb [hunk 2]
```diff
@@ -50,7 +50,9 @@
       <%= link_to "View business details", business_url(activity.business_id), class: "govuk-link psd-block-link" %>
     <% end %>
 
-    <% if activity.has_attachment? %>
+    <% if activity.is_a? AuditActivity::Test::Result %>
```
Comment (frankieroberto) [inline psd-web/app/views/investigations/activities/_activity_card.html.erb:53]: This type-check isn't ideal, but hopefully short-lived as we start to refactor the activity card HTML rendering.

**Comment by edwardhorsford** [comment]
Had a look - looking good so far.

A few things to clean up:

* Need to make sure all details collected are displayed.
** Further details not displayed
** Product tested not displayed
** Attachment description not displayed (this is a bit awkward as attachment is on the right, but I think for now we can include in table.
* Can we change `Date` to `Date of test`
* Link for `view test result` on the activity timeline would ideally be unique - I wonder if we can include some of the title in it?

File: psd-web/app/models/audit_activity/test/result.rb (hunk 1 above)
Comment (edwardhorsford) [reply psd-web/app/models/audit_activity/test/result.rb:26]: Should we raise a card to track this work?

File: psd-web/app/views/investigations/test_results/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          key: { text: "Product tested" },
+          value: { html: link_to(@test.product.name, product_path(@test.product)) }
+        },
+        {
+          key: { text: "Legislation" },
+          value: { text: @test.legislation }
+        },
+        {
+          key: { text: "Result" },
+          value: { text: @test.result.capitalize }
```
Comment (edwardhorsford) [inline psd-web/app/views/investigations/test_results/show.html.erb:38]:
```suggestion
          value: { text: @test.result.upcase_first }
```

File: psd-web/app/controllers/investigations/test_results_controller.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class Investigations::TestResultsController < ApplicationController
+  def show
+    @investigation = Investigation.find_by(pretty_id: params[:investigation_pretty_id])
+    authorize @investigation, :view_non_protected_details?
+    @test = @investigation.test_results.find(params[:id]).decorate
```
Comment (slawosz) [inline psd-web/app/controllers/investigations/test_results_controller.rb:5]: Rename to `@test_result`?

Comment (frankieroberto) [reply psd-web/app/controllers/investigations/test_results_controller.rb:5]: Good idea. 👍

**Comment by edwardhorsford** [comment]
A demo for Kam, this is `different`

**Pull request merged**